src/perftest_resources.c

```diff
@@ -3439,7 +3439,7 @@ int ctx_alloc_credit(struct pingpong_context *ctx,
 	int flags = IBV_ACCESS_LOCAL_WRITE | IBV_ACCESS_REMOTE_WRITE;
 	int i;
 
-	ALLOCATE(ctx->ctrl_buf,uint32_t,user_param->num_of_qps);
+	ALLOCATE(ctx->ctrl_buf,uint32_t,buf_size);
```

Comment:
Please take a look at #define ALLOCATE in perftest_parameters.h. So allocation of buf_size*sizeof(uint32_t) is too much. May be 2*user_param->num_of_qps is the correct value here...